internal/impl/kafka/enterprise/schema_registry_input.go

```diff
@@ -166,6 +160,12 @@ func inputFromParsed(pConf *service.ParsedConfig, mgr *service.Resources) (i *sc
 		return nil, fmt.Errorf("failed to create Schema Registry client: %s", err)
 	}
 
+	if label := mgr.Label(); label != "" {
+		mgr.SetGeneric(mgr.Label(), i)
```

Comment:
instead of a raw string can we use some typed wrapper so it's not possible for another component to clobber this? Then we don't need to do they empty string special case.